plugins/jobs/src/js/JobsTabContainer.js

```diff
@@ -0,0 +1,155 @@
+  searchString: ""
+};
+
+class JobsTabContainer extends mixin(StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.state = { ...DEFAULT_FILTER_OPTIONS };
+
+    this.store_listeners = [
+      { name: "dcos", events: ["change"], suppressUpdate: false },
+      {
+        name: "metronome",
+        events: ["change", "jobCreateSuccess"],
+        suppressUpdate: false
+      }
+    ];
+
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentDidMount() {
+    super.componentDidMount();
+    this.updateFromProps(this.props);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    super.componentWillReceiveProps(...arguments);
+    this.updateFromProps(nextProps);
+  }
+
+  handleFilterChange(filterValue) {
+    const { router } = this.context;
+    const {
+      location: { pathname }
+    } = this.props;
+    const query = { [JobFilterTypes.TEXT]: filterValue };
+
+    router.push({ pathname, query });
+
+    this.setState({
+      searchString: filterValue
+    });
+  }
+
+  resetFilterQueryParams() {
+    const {
+      location: { pathname }
+    } = this.props;
+    const query = Object.assign({}, location.query);
+
+    Object.values(JobFilterTypes).forEach(function(filterKey) {
+      delete query[filterKey];
+    });
+
+    this.context.router.push({ pathname, query });
+  }
+
+  resetFilter() {
+    var state = Object.assign({}, this.state, DEFAULT_FILTER_OPTIONS);
+    this.setState(state, this.resetFilterQueryParams);
+  }
+
+  updateFromProps(props) {
+    const { location = { query: {} } } = props;
+
+    if (location.query[JobFilterTypes.TEXT] != null) {
+      const state = {};
+
+      state[JobFilterTypes.TEXT] = location.query[JobFilterTypes.TEXT];
+      this.setState(state);
+    }
+  }
+
+  getFilteredJobs(item) {
+    const { searchString } = this.state;
+    let jobs = item.getItems();
+
+    if (searchString) {
+      const filterProperties = Object.assign({}, item.getFilterProperties(), {
+        name(item) {
+          return item.getId();
+        }
+      });
+
+      jobs = item.filterItemsByText(searchString, filterProperties).getItems();
+    }
+
+    return jobs;
+  }
+
+  render() {
+    const id = decodeURIComponent(this.props.params.id);
+    const item =
+      DCOSStore.jobTree.findItem(function(item) {
+        return item instanceof JobTree && item.id === id;
+      }) || DCOSStore.jobTree;
+    const isLoading = DCOSStore.jobDataReceived;
+    const root = DCOSStore.jobTree;
+    const filteredJobs = this.getFilteredJobs(item);
+    const searchString = this.state.searchString;
+    const handleFilterChange = this.handleFilterChange;
+    const resetFilter = this.resetFilter;
+    const hasFilterApplied = Object.keys(DEFAULT_FILTER_OPTIONS).some(
+      filterKey => {
+        return (
+          this.state[filterKey] != null && this.state[filterKey].length > 0
+        );
+      }
+    );
+
+    return isLoading ? (
```

Comment:
you sure you get a component here?

Reply:
yup, fixed the issue i had 👍

Reply:
🎨  I think the if isLoading return <...> is easier to read.

Reply:
I can change this, but then I'll add https://eslint.org/docs/rules/no-ternary to our eslint rules… It annoys me that this is something we allow but i get a comment _everytime_ I use it…

Reply:
Hum, I would not pull extreme to the other side. In my opinion on style, ternaries are good for short clauses, I try not to use them if it spans more than a single line. I would not be in favor of the death of all ternaries.

Reply:
+1 on @Fabs point of view

Reply:
this _is_ a short clause 🤔

Reply:
I dont think this is a short clause. Short is a single line, with probably the predicate being a variable or a very simple condition.

```js
return isLoading ? jobsTabLoading : JobsTab;
```
That I would be fine with. But I won`t block it because of that.

Reply:
I'm fine with this ternary.

Reply:
Ok, short for me is `return (clause) ? A : B;`. I dont really care if `A = true` or `A = <Component with={} a={} lot={} of={} props={} />`.
Long for me would be something like `return (clauseA) ? (clauseAB) ? AB1 : AB2 : (clauseAC) ? AC1 : AC2;` 

whatever, please lets stop commenting this on every single PR if we dont add the rule 🙏

Reply:
I also receive comments about style from time to time, that I disagree with. Nonetheless, I think we should not shy about sharing our perceptions on style.

The ternary on this piece of code does not match my bar for `short`, I would also not be in favor of banning ternaries for life.

I approved the PR nonetheless ✅

Reply:
I might be overreacting here, but I received comments like yours almost _everytime_ I used ternaries - if it is such a bad thing, we should ban it, if not, please stop complaining (the team as a whole, not you in particular). nvm - thanks for approving 🙇

Reply:
I'd argue that the usage of a ternary here makes it hard to follow the flow and would say that an early exit would probably be easier to follow, but agree with the general sentiment that we shouldn't manually enforce styling.

The problem here is that there is no eslint rule for this case only; thus we either need to write one or would have to ban ternaries in general. Prohibiting ternaries altogether would bloat cases like the following and is therefore not desirable. 

```jsx
 <span>
        {predicate ? a : b}
 </span>
```
Thus we should implement a rule for cases like this one and/or accept that there are always cases in which we will discuss and manually enforce style as not everything that is theoretically "allowed" is also desirable.